packages/drivers/odsp-driver/src/compactSnapshotParser.ts

```diff
@@ -74,6 +90,56 @@ function readTreeSection(node: NodeCore) {
     };
     for (const treeNode of node) {
         assertNodeCoreInstance(treeNode, "tree nodes should be nodes");
+
+        /**
+         * Perf optimization - the most common cases!
+         * This is essentially unrolling code below for faster processing
+         * It speeds up tree parsing by 2-3x times!
+         */
+        if (treeNode.getMaybeString(0) === "name") {
+            const length = treeNode.length;
+            if (length === 4) {
+                const content = treeNode.getMaybeString(2);
+                // "name": <node name>
+                // "children": <blob id>
+                if (content === "children") {
+                    trees[treeNode.getString(1)] = readTreeSection(treeNode.getNode(3));
+                    continue;
+                }
+                // "name": <node name>
+                // "value": <blob id>
+                if (content === "value") {
+                    snapshotTree.blobs[treeNode.getString(1)] = treeNode.getString(3);
+                    continue;
+                }
+            }
+
+            // "name": <node name>
+            // "nodeType": 3
+            // "value": <blob id>
+            if (length === 6 &&
```

Comment:
@jatgarg, can you please follow up with ODSP folks on what is this "nodeType" about? Why it's here, and can we remove it from payloads? This will make files smaller and will also allow us to remove this unneeded case from code.

Reply:
Opened https://dev.azure.com/fluidframework/internal/_workitems/edit/1993 to follow up

Reply:
Yeah it is something they use. I am not sure at the top of my head but we can ask.